sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverterTest.java

```diff
@@ -57,137 +65,159 @@
-              new DoFn<Row, Void>() {
-                @ProcessElement
-                public void processElement(ProcessContext context) {}
-              }));
-      return PDone.in(input.getPipeline());
+    @Test
+    public void testToEnumerable_count() {
+      Schema schema = Schema.builder().addInt64Field("id").build();
+      RelDataType type = CalciteUtils.toCalciteRowType(schema, TYPE_FACTORY);
+      ImmutableList<ImmutableList<RexLiteral>> tuples =
+          ImmutableList.of(
+              ImmutableList.of(rexBuilder.makeBigintLiteral(BigDecimal.ZERO)),
+              ImmutableList.of(rexBuilder.makeBigintLiteral(BigDecimal.ONE)));
+      BeamRelNode node =
+          new BeamIOSinkRel(
+              cluster,
+              RelOptTableImpl.create(null, type, ImmutableList.of(), null),
+              null,
+              new BeamValuesRel(cluster, type, tuples, null),
+              null,
+              null,
+              null,
+              false,
+              new FakeTable(),
+              null);
+
+      Enumerable<Object> enumerable = BeamEnumerableConverter.toEnumerable(options, node);
+      Enumerator<Object> enumerator = enumerable.enumerator();
+
+      assertTrue(enumerator.moveNext());
+      assertEquals(2L, enumerator.current());
+      assertFalse(enumerator.moveNext());
+      enumerator.close();
     }
   }
 
-  @Test
-  public void testToEnumerable_count() {
-    Schema schema = Schema.builder().addInt64Field("id").build();
-    RelDataType type = CalciteUtils.toCalciteRowType(schema, TYPE_FACTORY);
-    ImmutableList<ImmutableList<RexLiteral>> tuples =
-        ImmutableList.of(
-            ImmutableList.of(rexBuilder.makeBigintLiteral(BigDecimal.ZERO)),
-            ImmutableList.of(rexBuilder.makeBigintLiteral(BigDecimal.ONE)));
-    BeamRelNode node =
-        new BeamIOSinkRel(
-            cluster,
-            RelOptTableImpl.create(null, type, ImmutableList.of(), null),
-            null,
-            new BeamValuesRel(cluster, type, tuples, null),
-            null,
-            null,
-            null,
-            false,
-            new FakeTable(),
-            null);
-
-    Enumerable<Object> enumerable = BeamEnumerableConverter.toEnumerable(options, node);
-    Enumerator<Object> enumerator = enumerable.enumerator();
-
-    assertTrue(enumerator.moveNext());
-    assertEquals(2L, enumerator.current());
-    assertFalse(enumerator.moveNext());
-    enumerator.close();
+  /**
+   * Tests for BeamEnumerableConverter parameterized by field type. Only primitive field types are
+   * included.
+   */
+  @RunWith(Parameterized.class)
+  public static class PrimitiveTypeTests {
+    @Parameterized.Parameter public FieldType fieldType;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Iterable<Object[]> data() {
+      return Arrays.stream(TypeName.values())
+          .filter(TypeName::isPrimitiveType)
+          .map(typeName -> new Object[] {FieldType.of(typeName)})
```

Comment:
Could we also mock logical types of BeamSQL here? BeamSQL logical types are defined in https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java? BeamSQL uses logical type to define some of its types: Date, Time, Char, etc.

Reply:
Sure! Updated the test to list the logical types from CalciteUtils

Reply:
Do I miss something? Seems no change after last time?

Reply:
Doh! Sorry  I just force pushed the exact same commit :man_facepalming: check it now